projects/packages/forms/tests/php/contact-form/Feedback_Test.php

```diff
@@ -1773,4 +1773,43 @@ public function test_has_field_type_legacy_feedback() {
 		$this->assertTrue( $response->has_field_type( 'consent' ), 'Legacy feedback should report consent field exists' );
 		$this->assertFalse( $response->has_consent(), 'Legacy consent should default to false (no)' );
 	}
+
+	public function test_get_field_by_id_and_value_by_id_new_submission() {
+		$form_id    = Utility::get_form_id();
+		$_post_data = Utility::get_post_request(
+			array(
+				'name'    => 'John Doe',
+				'email'   => 'john@example.com',
+				'message' => 'Hello!',
+			),
+			'g' . $form_id
+		);
+
+		$form = new Contact_Form(
+			array(
+				'title'       => 'Test Form',
+				'description' => 'This is a test form.',
+			),
+			"[contact-field label='Name' type='name' required='1'/][contact-field label='Email' type='email' required='1'/][contact-field label='Message' type='textarea' required='1'/]"
+		);
+
+		$response  = Feedback::from_submission( $_post_data, $form );
+		$field_ids = $form->get_field_ids();
+		$email_id  = $field_ids['email'];
+
+		$this->assertNotEmpty( $email_id );
+		$this->assertEquals( 'john@example.com', $response->get_field_value_by_id( $email_id ) );
+
+		$field = $response->get_field_by_id( $email_id );
+		$this->assertInstanceOf( Feedback_Field::class, $field );
+		$this->assertEquals( $email_id, $field->get_form_field_id() );
```

Comment:
It would also be good to check that saving worked as expected and that the field_id would also be there after we save the response. 

So 
```
$saved_id = $response->save();
$saved_response = Feedback::get( $saved_id );
$this->assertEquals( 'john@example.com', $saved_response->get_field_value_by_id( $email_id ) );
```